### ISSUE TITLE
Suggestions welcome:  loading indicator on  a custom toolbar?

### ISSUE THREAD

**Issue by comrade2sun** [openseadragon/openseadragon#1039]
#### _Prolog_

_I added a little loading indicator to a custom toolbar while loading a set of new `tileSources` plus a new `background-image` for the OSD div. For now it indicates the loading process of the new background images, which is already sufficient for my purpose. 
Experimental example:_
<a href="http://homepage.o2mail.de/nixneues/cojp/utamaro.html">Kitagawa Utamaro - Gallery</a>

_Even it might be a bit of an over bloated use case, **but...**_
#### The idea ✨

Show a loading indicator everytime new tiles are loaded.
...
The Script I am using right now:

``` javascript
var progBar = {template:3,parent:'#toolbarDiv'};
var queryProgress = new Mprogress(progBar);
    // bunch of other vars
var viewer = new OpenSeadragon.Viewer({ //  option, create viewer etc.
    // load image set 1
$('#pg01').click ... // with 1 tileSource
    // load image set 2
$('#pg02').click(function () {
    viewer.open([{x : -0.55,y : 0.25,height : 1,tileSource : yogoto}, 
    // plus 3 more tileSources
        ]);
    // change background-image
    $('.openseadragon').removeClass().addClass('openseadragon conta bg-2');
    // start the progressbar
    queryProgress.start();
    // stop the progressbar
    $('#osd-canvas').imagesLoaded({
        background : true
    }, function () {
        queryProgress.end();
    });
});
    // load image set 3
$('#pg03').click ...  $('#pg04').click ... and so on
    //  other script
/*
Dependencies for loading indicator:
MProgress.js: https://github.com/lightningtgc/MProgress.js
imagesloaded: https://github.com/desandro/imagesloaded
*/
```
#### THE UNKNOWN

How to start and stop the loading indicator for newly loaded tiles, similar as in the background-image case?
#### The Question

Is THIS worthwhile digging in deeper? <i>... anyway I might learn something else about OSD</i> 😇 
#### State of the art

My JavaScript skills are zilch + 0.1%  <i>(Please be patient with me!)</i>
...
Any suggestion and/or criticism from experienced JavaScript Wizards are welcome. 🌴 

Thanks for reading

**Comment by avandecreme** [comment]
It looks like you need to listen to the new `fully-loaded` event.
It has been implemented in #837 and is currently only available in the master branch.
Let me know if you need any help with it.

**Comment by comrade2sun** [comment]
Thanks for your reply

Because this idea is more a luxury question, it doesn't matter if I find a solution sooner or later (or never) ...
... but I would appreciate your help. 
I am not in a hurry and so please take your time.

Let me dig in #837 discussion first and see if understand any of it.
